### PR TITLE
OnlineDDL: implementing -postpone-completion, ALTER VITESS_MIGRATION ... COMPLETE

### PR DESCRIPTION
**Pull request by shlomi-noach**
This PR implements:

## -postpone-completion
`-postpone-completion` flag in `ddl_strategy`. This flag indicates that the migration should not auto-complete. This applies for:
- any `CREATE TABLE`
- any `DROP TABLE`
- `ALTER` table in `online` strategy
- `ALTER` table in `gh-ost` strategy

Note that this flag is not supported for `pt-osc` strategy.

Behavior of migrations with this flag:
- an `ALTER` table begins, runs, but does not cut-over.
- `CREATE` or `DROP` migrations are silently not even scheduled

Complimenting this flag is:

## ALTER VITESS_MIGRATION '-uuid-' COMPLETE

This command indicates that a migration executed with `-postpone-completion` is good to complete. Behavior:

- For running `ALTER`s (`online` and `gh-ost`) which are ready to cut-over: cut-over imminently (though not immediately - cut-over depends on polling interval, replication lag, etc)
- For running `ALTER`s (`online` and `gh-ost`) which are only partly through the migration: they will cut-over automatically when they complete their work, as if `-postpone-completion` wasn't indicated
- For queued `CREATE` and `DROP` migrations: "unblock" them from being scheduled. They'll be scheduled at the schedulers discretion. there is no guarantee that they will be scheduled to run immediately.
---

tests added:

- `vrepl` endtoend
- `gh-ost` endtoend
---

This PR extends https://github.com/vitessio/vitess/pull/9160 (not yet merged at this time)
Tracking issue: #6926

### CODE REVIEW

**Comment by shlomi-noach** [comment]
As #9160 is merged, this PR is now ready to review. /cc @deepthi @sougou

File: go/vt/vttablet/onlineddl/executor.go [hunk 1]
```diff
@@ -1561,6 +1581,75 @@ func (e *Executor) scheduleNextMigration(ctx context.Context) error {
 	return err
 }
 
+// reviewQueuedMigrations iterates queued migrations and sees if any information needs to be updated
+func (e *Executor) reviewQueuedMigrations(ctx context.Context) error {
+	e.migrationMutex.Lock()
+	defer e.migrationMutex.Unlock()
+
+	{
```
Comment (frouioui) [inline go/vt/vttablet/onlineddl/executor.go:1589]: Can we remove this new block? I am maybe missing its added value

**Review by frouioui — APPROVED** [review]
I do not have a lot of context on OnlineDDL, but the changes look good to me. We should wait for someone with more context to review this too :)

File: go/vt/vttablet/onlineddl/executor.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/vttablet/onlineddl/executor.go:1589]: Removed! I was thinking there could be multiple "sections" for reviewing the migrations, and so each would be in a separate block, for clarity. But right now there was just this one...

Comment (frouioui) [reply go/vt/vttablet/onlineddl/executor.go]: Okay, I understand now

**Comment by shlomi-noach** [comment]
Noting that `-postpone-completion` implementation is incomplete, as it will withhold `-declarative` migrations; this will be addressed in a separate PR.

File: doc/releasenotes/13_0_0_summary.md [hunk 2]
```diff
@@ -1,16 +1,48 @@
...
 ### alter vitess_migration ... cleanup
+
 A new query is supported:
+
 ```sql
 alter vitess_migration '9748c3b7_7fdb_11eb_ac2c_f875a4d24e90' cleanup
 ```
-This query tells Vitess that a migration's artifacts are good to be cleaned up asap. This allows Vitess to free disk resources sooner. As a reminder, once a migration's artifacts are cleaned up, the migration is no
+
+This query tells Vitess that a migration's artifacts are good to be cleaned up asap. This allows Vitess to free disk resources sooner. As reminder, once a migration's artifacts are cleaned up, the migration is no
```
Comment (deepthi) [inline doc/releasenotes/13_0_0_summary.md:29]:
```suggestion
This query tells Vitess that a migration's artifacts are good to be cleaned up asap. This allows Vitess to free disk resources sooner. As a reminder, once a migration's artifacts are cleaned up, the migration is no
```

File: doc/releasenotes/13_0_0_summary.md [hunk 3]
```diff
@@ -1,16 +1,48 @@
...
+
+```sql
+alter vitess_migration '9748c3b7_7fdb_11eb_ac2c_f875a4d24e90' complete
+```
+
+This command indicates that a migration executed with `-postpone-completion` is good to complete. Behavior:
+
+- For running `ALTER`s (`online` and `gh-ost`) which are ready to cut-over: cut-over imminently (though not immediately - cut-over depends on polling interval, replication lag, etc)
+- For running `ALTER`s (`online` and `gh-ost`) which are only partly through the migration: they will cut-over automatically when they complete their work, as if `-postpone-completion` wasn't indicated
+- For queued `CREATE` and `DROP` migrations: "unblock" them from being scheduled. They'll be scheduled at the schedulers discretion. there is no guarantee that they will be scheduled to run immediately.
```
Comment (deepthi) [inline doc/releasenotes/13_0_0_summary.md:44]:
```suggestion
- For queued `CREATE` and `DROP` migrations: "unblock" them from being scheduled. They'll be scheduled at the scheduler's discretion. there is no guarantee that they will be scheduled to run immediately.
```

**Review by deepthi — APPROVED** [review]
A couple of documentation nits, otherwise LGTM.
It's a pleasure to review a PR that is more tests than code :100:

File: doc/releasenotes/13_0_0_summary.md (hunk 2 above)
Comment (shlomi-noach) [reply doc/releasenotes/13_0_0_summary.md]: fixed

File: doc/releasenotes/13_0_0_summary.md (hunk 3 above)
Comment (shlomi-noach) [reply doc/releasenotes/13_0_0_summary.md]: fixed

**Comment by shlomi-noach** [comment]
Dcoumentation PR: https://github.com/vitessio/website/pull/887

**Pull request merged**